client/src/components/AdminPanel/ManageMonuments/ManageMonument/ManageMonument.js

```diff
@@ -0,0 +1,132 @@
+        this.state = {
+            toggleActiveModalOpen: false,
+            deleteModalOpen: false
+        };
+    }
+
+    toggleActive(active) {
+        const { onToggleActive } = this.props;
+        onToggleActive(active);
+        this.setState({toggleActiveModalOpen: false});
+    }
+
+    deleteMonument() {
+        const { onDeleteMonument } = this.props;
+        onDeleteMonument();
+        this.setState({deleteModalOpen: false});
+    }
+
+    render() {
+        const { monument, deleted } = this.props;
+        if (deleted) return (
+            <div className="manage-monument">
+                <div>
+                    You have deleted {monument.title} successfully.
+                </div>
+            </div>
+        );
+        return (<>
+            {this.renderManageMonument()}
+            {this.renderConfirmToggleActiveModal()}
+            {this.renderConfirmDeleteModal()}
+        </>)
+    }
+
+    renderManageMonument() {
+        const { monument } = this.props;
+
+        return (
+            <div className="manage-monument">
+                <About monument={monument} contributions={monument.contributions} references={monument.references}
+                       header={monument.title} showHiddenFields hideExportToCSV hideTitle/>
+                {(monument.images && monument.images.length) ?
+                    <div className="images">
+                        {monument.images.map(image => (
+                            <div key={image.id} className="image" style={{backgroundImage: `url(${image.url})`}}/>
+                        ))}
+                    </div> : <></>
+                }
+                {!monument.isActive &&
+                <Alert variant="info">
+                    This Monument or Memorial is marked as inactive, which means the public cannot view it. Press the
+                    activate button if you'd like to make it publicly visible.
+                </Alert>
+                }
+                <div className="buttons">
+                    {monument.isActive &&
+                    <Link to={`/monuments/${monument.id}`} className="btn btn-light">View Public Page</Link>
+                    }
+                    <Link to={`/panel/manage/monuments/monument/update/${monument.id}`} className="btn btn-light">Edit</Link>
+                    <Button variant="light" onClick={() => this.setState({toggleActiveModalOpen: true})}>{monument.isActive ? 'Deactivate' : 'Activate'}</Button>
+                    <Button variant="danger" onClick={() => this.setState({deleteModalOpen: true})}>Delete</Button>
+                </div>
+            </div>
+        );
+    }
+
+    renderConfirmToggleActiveModal() {
+        const { toggleActiveModalOpen } = this.state;
+        const { monument } = this.props;
+
+        return (
+            <div onClick={e => e.stopPropagation()}>
+                <Modal show={toggleActiveModalOpen} onHide={() => this.setState({toggleActiveModalOpen: false})}>
```

Comment:
The `onHide` here is clean. Wish I would have done that more instead if creating another function for it.